packages/serverpod/test/server/middleware/middleware_export_test.dart

```diff
@@ -0,0 +1,144 @@
+import 'package:serverpod/serverpod.dart';
+import 'package:test/test.dart';
+
+/// Tests that verify middleware-related types and implementations are properly
+/// exported from the serverpod.dart public API.
+///
+/// These tests ensure that users can access all necessary middleware functionality
+/// without needing to import internal implementation files.
+void main() {
+  group('Middleware exports', () {
+    group('Relic middleware types', () {
+      test(
+        'Given middleware exports When accessing Middleware type Then it is available',
```

Comment:
nit.
```suggestion
        'Given middleware exports '
        'when accessing Middleware type '
        'then it is available',
```
So no capitalising of _When_ and _Then_ here, and possibly split across multiple lines, seems to be the preferred format in serverpod tests.